### DISCUSSION TITLE
Can someone tell me about user sessions?

### DISCUSSION THREAD

**Discussion by AhangarAamir** [open-webui/open-webui#4258] [Q&A]
Can someone tell me about user sessions? For example, if User A uses one browser to chat and then uses another browser, does this application handle the sessions for this user?

**Comment by GylanSalih** [comment]
so far i know In an application like Open WebUI, managing user sessions usually involves creating a unique session ID or token when you log in, which is stored in your browser’s cookies or local storage. When you switch to a different browser, this session information doesn’t automatically transfer over because each browser keeps its own session data. This means you’ll need to log in again on the new browser to start a new session.

If the application uses a central system to store session data, you might be able to pick up where you left off after logging in again from a different browser. However, you’ll need to re-authenticate to access your session. Some apps use features like persistent tokens or Single Sign-On (SSO) to make it easier to stay logged in across different browsers and devices. So, if Open WebUI doesn’t handle cross-browser sessions specifically, you’ll need to log in separately in each browser.

**Reply by AhangarAamir** [reply]
@GylanSalih thank you for your reply

**Reply by nomppy** [reply]
when you login a jwt token is created by OI and stored in both your browser's localStorage and cookies as an httpOnly cookie. as long as the token exists in localStorage you will be signed in automatically. you can configure the expiration time of this token with a environmental variable

**Reply by GylanSalih** [reply]
> @GylanSalih thank you for your reply

No problem just marked as solved if its solved your problem, would also help my profil haha

**Reply by thiswillbeyourgithub** [reply]
> when you login a jwt token is created by OI and stored in both your browser's localStorage and cookies as an httpOnly cookie. as long as the token exists in localStorage you will be signed in automatically. you can configure the expiration time of this token with a environmental variable

Hi, @nomppy , when im restarting the docker compose container I have to relog every session. Do you know what container folder I should bind to a persistent folder to make it remember these sessions?

**Reply by thiswillbeyourgithub** [reply]
Apparently fixed by #5131
